File: backend/TenderGo.Models/Requests/AdminResetPasswordRequest.cs
using System.ComponentModel.DataAnnotations;

namespace TenderGo.Api.Controllers
{
    public class AdminResetPasswordRequest
    {
        [Required]
        [MinLength(8, ErrorMessage = "Password has to be 8 characters minimum.")]
        [RegularExpression(@"^(?=.*[a-z])(?=.*[A-Z])(?=.*\d).+$", ErrorMessage = "Password must contain a capital letter, a small letter, and a number.")]
        public string NewPassword { get; set; }

        [Required]
        [Compare("NewPassword", ErrorMessage = "Passwords do not match.")]
        public string ConfirmPassword { get; set; }
    }
}

File: UI/tendergo_mobile/lib/shared/models/requests/admin_reset_password_request.dart
class AdminResetPasswordRequest {
  final String newPassword;
  final String confirmPassword;

  const AdminResetPasswordRequest({
    required this.newPassword,
    required this.confirmPassword,
  });

  Map<String, dynamic> toJson() =>{
    'newPassword': newPassword,
    'confirmPassword': confirmPassword,
  };
}